### PR TITLE
flutter test --start-paused

### LINKED ISSUE #163: Debugging tests is unnecessarily hard
<a href="https://github.com/Hixie"></img></a> **Issue by [Hixie](https://github.com/Hixie)**
_Wednesday Sep 16, 2015 at 19:49 GMT_
_Originally opened as https://github.com/flutter/engine/issues/1196_

---

It would be great if there was a way to launch a single test in Observatory. Right now since there's no way to set breakpoints before you've loaded the code, the only way I can think to do it is to modify the code to call debugger(), then run the test, then attach the debugger (somewhere? it's not clear where to attach it, really). Really I want the equivalent of "start with gdb" but for observatory.

### PR DESCRIPTION
**Pull request by Hixie**
Make debugging tests in a debugger easier.

Fixes https://github.com/flutter/flutter/issues/163

### CODE REVIEW

**Comment by Hixie** [comment]
@tvolkert

File: packages/flutter_tools/lib/src/test/flutter_platform.dart [hunk 1]
```diff
@@ -70,12 +72,13 @@ enum _TestResult { crashed, harnessBailed, testBailed }
 typedef Future<Null> _Finalizer();
 
 class FlutterPlatform extends PlatformPlugin {
-  FlutterPlatform({ this.shellPath, this.collector, this.explicitObservatoryPort, this.explicitDiagnosticPort }) {
+  FlutterPlatform({ this.shellPath, this.collector, this.debuggerMode, this.explicitObservatoryPort, this.explicitDiagnosticPort }) {
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/test/flutter_platform.dart]: We should probably make this constructor private

File: packages/flutter_tools/lib/src/test/flutter_platform.dart [hunk 2]
```diff
@@ -51,6 +51,7 @@ final InternetAddress _kHost = InternetAddress.LOOPBACK_IP_V4;
 void installHook({
   @required String shellPath,
   CoverageCollector collector,
+  bool debuggerMode,
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/test/flutter_platform.dart]: Make it default to false

**Comment by tvolkert** [comment]
LGTM

**Pull request merged**